src/boot/efi/linux_x86.c

```diff
@@ -107,6 +107,10 @@ static void linux_efi_handover(EFI_HANDLE image, uintptr_t kernel, BootParams *p
 
         kernel += (params->hdr.setup_sects + 1) * KERNEL_SECTOR_SIZE; /* 32bit entry address. */
 
+        /* Old kernels needs this set, while newer ones seem to ignore this. Note that this gets truncated on
+         * above 4G boots, which is fine as long as we do not use the value to jump to kernel entry. */
+        params->hdr.code32_start = kernel;
```

Comment:
shouldn't we do the truncation explicitly? i.e. `kernel & UINT32_MAX` or so?

comment should probably mean that 4.18 is the cutoff point.

Reply:
I see little point in doing so when the comment is already mentioning truncation.

And I have no idea where the cutoff point is. I only know that 4.18 at least needs it.